frontend/src/features/env/AdminFeedbackModal.tsx

```diff
@@ -0,0 +1,161 @@
+// TODO #4279: Remove after React rollout is complete
+import { useCallback, useRef } from 'react'
+import { Controller, useForm } from 'react-hook-form'
+import {
+  chakra,
+  FormControl,
+  Input,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Stack,
+  useBreakpointValue,
+} from '@chakra-ui/react'
+import { datadogRum } from '@datadog/browser-rum'
+import { isEmpty } from 'lodash'
+
+import { AdminFeedbackFormDto } from '~shared/types'
+
+import { useIsMobile } from '~hooks/useIsMobile'
+import { useToast } from '~hooks/useToast'
+import Button from '~components/Button'
+import Rating from '~components/Field/Rating'
+import FormErrorMessage from '~components/FormControl/FormErrorMessage'
+import FormLabel from '~components/FormControl/FormLabel'
+import { ModalCloseButton } from '~components/Modal'
+import Textarea from '~components/Textarea'
+
+import { useUser } from '~features/user/queries'
+
+import { useAdminFeedbackMutation } from './mutations'
+
+export const AdminFeedbackModal = ({
+  isOpen,
+  onClose,
+}: {
+  isOpen: boolean
+  onClose: () => void
+}): JSX.Element => {
+  const modalSize = useBreakpointValue({
+    base: 'mobile',
+    xs: 'mobile',
+    md: 'md',
+  })
+  const isMobile = useIsMobile()
+  const toast = useToast({ status: 'success', isClosable: true })
+
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm<AdminFeedbackFormDto>()
+
+  const initialRef = useRef(null)
+
+  const { user } = useUser()
+  const url = window.location.href
+  const rumSessionId = datadogRum.getInternalContext()?.session_id
+
+  const adminFeedbackMutation = useAdminFeedbackMutation()
+
+  const handleSubmitForm = handleSubmit((inputs: AdminFeedbackFormDto) => {
+    adminFeedbackMutation.mutateAsync(inputs)
+    toast({ description: 'Your feedback has been submitted.' })
+    onClose()
+  })
+
+  const handleClose = useCallback(() => {
+    onClose()
+  }, [onClose])
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={handleClose}
+      size={modalSize}
+      initialFocusRef={initialRef}
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalCloseButton />
+        <chakra.form noValidate onSubmit={handleSubmitForm}>
+          <ModalHeader pr="48px">Share your feedback</ModalHeader>
+          <ModalBody mt="0" pt="0">
+            <Stack spacing="1rem">
+              <FormControl>
+                <Input type="hidden" {...register('url')} value={url} />
+              </FormControl>
+
+              <FormControl isRequired isInvalid={!isEmpty(errors)}>
+                <FormLabel>How was your experience using FormSG?</FormLabel>
+                <Controller
+                  rules={{ required: 'This field is required' }}
+                  control={control}
+                  name={'rating'}
```

Comment:
supernit: 
```suggestion
                  name="rating"
```

edit: and all the rest haha don't want to spam PR review with supernit comments